pkg/util/validation/stack.go

```diff
@@ -28,7 +28,17 @@ func validateStackName(s string) error {
 	if stackNameRE.MatchString(s) {
 		return nil
 	}
-	return errors.New("a stack name may only contain alphanumeric, hyphens, underscores, or periods")
+	return errors.New("stack names may only contain alphanumeric, hyphens, underscores, or periods")
+}
+
+// validateStackTagName checks if s is a valid stack tag name, otherwise returns a descriptive error.
+// This should match the stack naming rules enforced by the Pulumi Service.
+func validateStackTagName(s string) error {
+	tagNameRE := regexp.MustCompile("^[a-zA-Z0-9-_.:]{1,40}$")
```

Comment:
Just to keep ourselves honest, let's add some negative unit tests that `validateStackTagName` will reject, such as "foo***bar"... it could be the case that the period `.` in the `[...]` syntax doesn't mean what we think it means, and we instead need to escape it like `\.` to mean literally a period. (And a quick unit test would quickly confirm/disprove that.)